settings: Show indicator on notification settings.

vighneshbhat9945 opened the pull request:
This PR  fix this issue by showing indicator on notification settings heading.

Fixes: #28728.

<h3>Before Changes:</h3>

![before7](https://github.com/zulip/zulip/assets/153224120/2d6bff2a-3e7a-4ca6-8c22-2be4dd06fa2c)

<h3>After Changes:</h3>

![after7](https://github.com/zulip/zulip/assets/153224120/d0ebc2b2-38a6-442e-8dbc-348b65153b2f)

alya commented:
The screen capture looks good to me!

alya commented:
Please complete the self-review checklist, and post a comment to request a review from @sahil839  when ready.

vighneshbhat9945 commented:
Please  @sahil839  review my PR .

sahil839 commented on web/src/stream_edit.js:
```diff
@@ -319,6 +319,13 @@ function stream_setting_changed(e) {
+    } else {
+        $status_element = $(
+            `#stream_change_notification_setting_status${CSS.escape(sub.stream_id)}`,
+        );
```
It would be better if just not use this `if` condition and instead find the element using `.closest(".alert-notification")`. This way the status element would not need ID attribute as well.

vighneshbhat9945 replied:
Thanks @sahil839  I will implement the PR  using  ```.closest(".alert-notification").```
Before  that  I have found that   ```stream_is_muted_changed()```  uses  repeated logic .
so  I have  updated the PR removing that function and  handled the  ```is_muted``` property in ```stream_setting_changed()``` everything is working fine in terms of output. please tell me , am I missing out  something?Is this correct?.

sahil839 replied:
I think that would work fine by reading the code but I have not tested it yet. Can do it once the PR is updated with all the required changes. But that change should be in its own separate commit with proper explaination in the commit message and not in the commit for fixing the indicator.

sahil839 reviewed (commented):
Posted one comment.

vighneshbhat9945 commented:
@sahil839   I have implemented all the things discussed above please review it .
To find element  using  ```.closest(".alert-notification)```  was not possible in the existing DOM structure. So I have changed a little bit.  
<b>structure before :</b>

![Screenshot (86)](https://github.com/zulip/zulip/assets/153224120/b4f5f242-0ba3-493f-b5a9-7af3b941afbc)

<b>structure after:</b>

![Screenshot (87)](https://github.com/zulip/zulip/assets/153224120/94808082-928a-4033-8e25-00d209dc047a)

sahil839 commented:
Yeah, we would need some changes in DOM structure, but I think we can do something better than what it is done currently in the PR.
- Can we remove `subscription-config` div? I don't think it is used. If we can would be good to do so in a separate commit.
- The `subsection-header` div should be inside `subsection-parent`. This way we would have the structure somewhat similar to display settings, you can have a look at `display_settings.hbs` file, and then you can get the selector using `.closest(".subsection-parent").find(".alert-notification")`.

I think you can try the above changes and make sure that the design remains similar to what we have in main. Feel free to ask any further questions.

vighneshbhat9945 commented:
@sahil839   I have updated   the PR  the DOM structure looks good now. Please  review.

Thewhite33 commented:
Can I work on this??

sahil839 commented:
Can you rebase the PR to fix the failing test which is being discussed [here](https://chat.zulip.org/#narrow/stream/43-automated-testing/topic/main.20failing)? Looks good to me otherwise.

vighneshbhat9945 commented:
@sahil839   PR  rebased.

sahil839 commented:
Looks good to me. @timabbott ready for your review.

timabbott commented:
@vighneshbhat9945 sorry for the slow review; @sahil839 forgot to add the "integration review" label that I track. Merged, thanks for doing this with a nice clean commit history! 

I encourage working on next issues in the settings/stream settings UI to use the expertise you've developed in this PR.

sahil839 commented:
Sorry, for missing adding the label.
